app/controllers/application_controller.rb

```diff
@@ -1,5 +1,13 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  protect_from_forgery with: :exception
+  before_action :update_allowed_parameters, if: :devise_controller?
+
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :email, :password) }
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:name, :surname, :email, :password, :current_password)
+    end
```

Comment:
- You have done well adding devise to your project, but your authentication is not working as expected, when I log in the app, it is expected I should sign in/ sign up before using the app but this doesn't work as expected. A possible reason for this is that you aren't authenticating the user before performing any actions. I suggest adding `authenticate_user!` to your before_action. Please fix this 🙏